tests/unit/policyEvaluator/requestUtils.spec.js

```diff
@@ -110,3 +111,79 @@ describe('requestUtils.getClientIp', () => {
         assert.strictEqual(result, dummyRemoteIP);
     });
 });
+
+describe('requestUtils.getHttpProtocolSecurity', () => {
+    const configWithProxy = require('../../utils/dummyS3ConfigProxy.json');
+    const configWithoutProxy = require('../../utils/dummyS3Config.json');
+    const testClientIp = '192.168.100.1';
+    const testProxyIp = '192.168.100.2';
+
+    it('should return true if request comes via trusted proxy with https proto header', () => {
+        const request = new DummyRequest({
+            headers: {
+                'x-forwarded-proto': 'https',
+            },
+            socket: {
+                remoteAddress: testProxyIp,
+            },
+        });
+        const result = requestUtils.getHttpProtocolSecurity(request, configWithProxy);
+        assert.strictEqual(result, true);
+    });
+
+    it('should return false if request comes via trusted proxy with http proto header', () => {
+        const request = new DummyRequest({
+            headers: {
+                'x-forwarded-proto': 'http',
+            },
+            socket: {
+                remoteAddress: testProxyIp,
+            },
+        });
+        const result = requestUtils.getHttpProtocolSecurity(request, configWithProxy);
+        assert.strictEqual(result, false);
+    });
+
+    it('should check TLS when request not from trusted proxy', () => {
+        const request = new DummyRequest({
+            headers: {
+                'x-forwarded-proto': 'https',
```

Comment:
we want to test here that this field is ignored : so it should be http...

either we have a TLS socket & x-forewarded-proto = http, or the opposite (kind of more logical, as it actually tests _do not trust x-forwarded-... field which do not come from a trusted proxy_)